From 8854a7580568914ee1f3eb384b25f239686deda3 Mon Sep 17 00:00:00 2001
From: Zhongmin Wu <zhongmin.wu@intel.com>
Date: Fri, 22 Feb 2019 14:34:52 +0800
Subject: [PATCH 1037/1214] [WA]: drm/i915: raise priority for atomic_commit

Some observation indicates the atomic work is occasionally scheduled
later than expected, and causing frame drop.

Tracked-On: OAM-76482
Signed-off-by: Zhongmin Wu <zhongmin.wu@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index d5d978b..ef05540 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -12962,7 +12962,7 @@ static int intel_atomic_commit(struct drm_device *dev,
 	if (nonblock && intel_state->modeset) {
 		queue_work(dev_priv->modeset_wq, &state->commit_work);
 	} else if (nonblock) {
-		queue_work(system_unbound_wq, &state->commit_work);
+		queue_work(system_highpri_wq, &state->commit_work);
 	} else {
 		if (intel_state->modeset)
 			flush_workqueue(dev_priv->modeset_wq);
-- 
2.7.4

